src/ess/amor/tools.py

```diff
@@ -1,67 +1,44 @@
-    :type q_d: scipp._scipp.core.Variable
+    Generate bin edges for a `q_grid` inspired by the custom linear-log grid from
+    Jochen Stahn at PSI.
 
-    :return: The bin edges to be used in q-binning.
-    :rtype: scipp._scipp.core.Variable
+    :param edges: The edges for the different parts of the mesh.
+    :param scale: A string or list of strings specifying the scaling for the different
+        parts of the mesh. Possible values for the scaling are `"linear"` and `"log"`.
+        If a list is supplied, the length of the list must be one less than the length
+        of the `edges` parameter.
+    :param num: An integer or a list of integers specifying the number of points to use
+        in each part of the mesh. If a list is supplied, the length of the list must be
+        one less than the length of the `edges` parameter.
     """
-    q_min = sc.to_unit(q_min, q_max.unit)
-    q_fix = sc.to_unit(q_fix, q_max.unit)
-    if d_q is None:
-        d_q = 0.05 * q_fix
-    if (q_min < q_fix).value and (q_fix < q_max).value:
-        n_linear = ((q_fix - q_min) / d_q + 0.5).astype(sc.dtype.int64)
-        q_linear = q_fix - d_q * sc.arange('qz', n_linear.value, 0, -1)
-        n_log = (sc.log(q_max / q_fix) / sc.log(1. + d_q / q_fix) + 0.5).astype(
-            sc.dtype.int64)
-        q_log = sc.Variable(
-            dims=['qz'],
-            values=(q_fix.value *
-                    (1. + d_q / q_fix).value**sc.arange('qz', 0,
-                                                        (n_log + 1).values, 1).values),
-            unit=q_max.unit)
-        return sc.concatenate(q_linear, q_log, 'qz')
-    elif (q_min < q_max).value and (q_max < q_fix).value:
-        n_linear = ((q_fix - q_min) / d_q + 0.5).astype(sc.dtype.int64)
-        m_linear = ((q_fix - q_max) / d_q - 0.5).astype(sc.dtype.int64)
-        q_linear = q_fix - d_q * sc.arange('qz', n_linear.value,
-                                           (m_linear - 1).value, -1)
-        return q_linear
-    elif (q_fix < q_min).value and (q_min < q_max).value:
-        n_log = (sc.log(q_max / q_fix) / sc.log(1. + d_q / q_fix) + 0.5).astype(
-            sc.dtype.int64)
-        m_log = (sc.log(q_min / q_fix) / sc.log(1. + d_q / q_fix) - 0.5).astype(
-            sc.dtype.int64)
-        q_log = sc.Variable(
-            dims=['qz'],
-            values=q_fix.value *
-            (1. + d_q / q_fix).values**sc.arange('qz', m_log.value,
-                                                 (n_log + 1).value, 1.).values,
-            unit=q_max.unit)
-        return q_log
-    else:
-        raise ValueError("The given values of input parameters are not compatible.")
+    if not isinstance(scale, list):
+        scale = [scale]
+    if not isinstance(num, list):
+        num = [num]
+    if len(scale) != len(edges) - 1:
+        raise ValueError("Sizes do not match")
```

Comment:
Can this error message be more specific?
```suggestion
        raise ValueError("Length of scale should be one greater than edges.")
```